packages/dashboard-frontend/src/pages/UserPreferences/GitServicesTab/index.tsx

```diff
@@ -0,0 +1,258 @@
+    this.setState({ currentGitOauthIndex: rowIndex, isRevokeModalOpen: true });
+  }
+
+  private async deleteOauth(gitOauth: api.GitOauthProvider): Promise<void> {
+    try {
+      await this.props.deleteOauth(gitOauth);
+      this.showAlert({
+        key: 'delete-github',
+        variant: AlertVariant.success,
+        title: `Git oauth '${gitOauth}' successfully deleted.`,
+      });
+    } catch (e) {
+      this.showAlert({
+        key: 'delete-fail',
+        variant: AlertVariant.danger,
+        title: helpers.errors.getMessage(e),
+      });
+    }
+  }
+
+  private async onRevoke(gitOauth?: api.GitOauthProvider): Promise<void> {
+    this.setState({ isRevokeModalOpen: false, currentGitOauthIndex: -1 });
+    if (gitOauth === undefined) {
+      this.state.selectedItems.forEach(async gitOauth => {
+        await this.deleteOauth(gitOauth);
+      });
+      this.setState({ selectedItems: [] });
+    } else {
+      await this.deleteOauth(gitOauth);
+    }
+  }
+
+  private setRevokeModalStatus(isRevokeModalOpen: boolean): void {
+    if (this.state.isRevokeModalOpen === isRevokeModalOpen) {
+      return;
+    }
+    this.setState({ isRevokeModalOpen });
+  }
+
+  private handleRevoke(): void {
+    this.setState({ currentGitOauthIndex: -1, isRevokeModalOpen: true });
+  }
+
+  render(): React.ReactNode {
+    const { isLoading } = this.props;
+    const { isRevokeModalOpen, currentGitOauth, selectedItems } = this.state;
+    const columns = ['Name', 'Server'];
+    const rows =
+      this.state.gitOauth.map(provider => ({
+        cells: this.buildGitOauthRow(
+          providers[provider.name] ? providers[provider.name] : provider.name,
+          provider.endpointUrl,
+        ),
+        selected: selectedItems.includes(provider.name),
+      })) || [];
+    const actions = [
+      {
+        title: 'Revoke',
+        onClick: (event, rowIndex) => this.showOnRevokeGitOauthModal(rowIndex),
+      },
+    ];
+
+    return (
+      <React.Fragment>
+        <ProgressIndicator isLoading={isLoading} />
+        <PageSection>
+          {rows.length === 0 ? (
+            <EmptyState text="No Git Services" />
+          ) : (
+            <React.Fragment>
+              <RevokeGitServicesModal
+                selectedItems={selectedItems}
+                onCancel={() => this.setRevokeModalStatus(false)}
+                onRevoke={() => this.onRevoke(currentGitOauth)}
+                isOpen={isRevokeModalOpen}
+                gitOauth={currentGitOauth}
+              />
+              <Toolbar id="git-services-list-toolbar" className="pf-m-page-insets">
+                <ToolbarContent>
+                  <ToolbarItem>
+                    <Button
+                      variant={ButtonVariant.danger}
+                      isDisabled={this.state.selectedItems.length === 0}
+                      data-testid="bulk-delete-button"
+                      onClick={() => this.handleRevoke()}
```

Comment:
Misleading method name is used.

```suggestion
                      onClick={() => this.handleModalShow()}
```

Reply:
I have fixed it.